### PR TITLE
Create your own Enumerable

### PR DESCRIPTION
**Pull request by PrantoshB**
## Hello There 👋🏼

In this branch, I and my Coding Partner @tanveerisonline have:

- Created a class `MyList` that has an instance variable `@list`.
- In `MyList` implemented a method #each that yields successive members of @list and uses the MyEnumerable module.
- Created a module `MyEnumerable` that implements the following methods:
     - `#all?`
     - `#any?`
     - `#filter`
- Verified the solution

### Thanks 🍻

### CODE REVIEW

File: mylist.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    @list = []
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
+end
+
+# Create our list
+list = MyList.new(1, 2, 3, 4)
```
Comment (Davidosky007) [inline mylist.rb:16]: * When I try to verify your solution I got empty arrays back which isn't the expected result according to the project requirement, Kindly fix it 🙏 
image
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @PrantoshB @tanveerisonline 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* No linter errors 👍 
* Well written PR description 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_enumerable.rb [hunk 2]
```diff
@@ -3,19 +3,20 @@ def all?(&block)
...
     end
+    false
   end
 
   def filter?(&block)
     result = []
     each do |item|
-      result << item if block.call(item)
+      result << item if block.call(item) == true
     end
```
Comment (DeliceLydia) [inline my_enumerable.rb:20]: - [ ] Please fix the `filter` method currently it can't display `even` numbers
Screen Shot 2023-03-03 at 18 37 58

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights
- No linter errors
- Good communication while submitting 

 ## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by PrantoshB** [comment]
Hello @DeliceLydia , Thanks for reviewing my project.

Screenshot 2023-03-03 at 10 23 39 PM


I have fixed the filter method.

File: my_enumerable.rb (hunk 2 above)
Comment (Abenezer-Tilahun) [reply my_enumerable.rb]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @PrantoshB  @tanveerisonline 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git  -Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**